tests/test_export_onnx.py

```diff
@@ -0,0 +1,123 @@
+from detectron2.modeling import build_model
+from detectron2.utils.env import TORCH_VERSION
+from detectron2.utils.testing import (
+    get_sample_coco_image,
+    SLOW_PUBLIC_CPU_TEST,
+    is_onnx_installed)
+
+
+SUPPORTED_ONNX_OPSET = 14
+
+
+# TODO: this test requires manifold access, see: T88318502
+class TestONNXTracingExport(unittest.TestCase):
+    def setUp(self):
+        if not is_onnx_installed():
+            self.skipTest("ONNX is not installed")
+
+    def testMaskRCNNFPN(self):
+        def inference_func(model, images):
+            inputs = [{"image": image} for image in images]
+            inst = model.inference(inputs, do_postprocess=False)[0]
+            return [{"instances": inst}]
+
+        self._test_model_zoo_from_config_path(
+            "COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml", inference_func
+        )
+
+    @SLOW_PUBLIC_CPU_TEST
+    def testMaskRCNNC4(self):
+        def inference_func(model, image):
+            inputs = [{"image": image}]
+            return model.inference(inputs, do_postprocess=False)[0]
+
+        self._test_model_zoo_from_config_path(
+            "COCO-InstanceSegmentation/mask_rcnn_R_50_C4_3x.yaml", inference_func
+        )
+
+    @SLOW_PUBLIC_CPU_TEST
+    def testCascadeRCNN(self):
+        def inference_func(model, image):
+            inputs = [{"image": image}]
+            return model.inference(inputs, do_postprocess=False)[0]
+
+        self._test_model_zoo_from_config_path(
+            "Misc/cascade_mask_rcnn_R_50_FPN_3x.yaml", inference_func
+        )
+
+    # # bug fixed by https://github.com/pytorch/pytorch/pull/67734
+    @unittest.skipIf(TORCH_VERSION == (1, 10) and os.environ.get("CI"), "1.10 has bugs.")
+    def testRetinaNet(self):
+        def inference_func(model, image):
+            return model.forward([{"image": image}])[0]["instances"]
+
+        self._test_model_zoo_from_config_path(
+            "COCO-Detection/retinanet_R_50_FPN_3x.yaml", inference_func
+        )
+
+    def _test_model(self, model, inputs):
+        import onnx  # noqa: F401
+        f = io.BytesIO()
+        with torch.no_grad():
+            torch.onnx.export(
+                model,
+                inputs,
+                f,
+                training=torch.onnx.TrainingMode.EVAL,
+                operator_export_type=torch.onnx.OperatorExportTypes.ONNX,
+                opset_version=SUPPORTED_ONNX_OPSET,
+            )
+        assert onnx.load_from_string(f.getvalue())
+
+    def _test_model_zoo_from_config_path(self, config_path, inference_func, batch=1):
+        # TODO: Using device='cuda' raises
+        #   RuntimeError: isBool() INTERNAL ASSERT FAILED at
+        #   "/github/pytorch/torch/include/ATen/core/ivalue.h":590,
+        #   please report a bug to PyTorch.
+        #   from /github/pytorch/torch/_ops.py:142 (return of OpOverloadPacket.__call__)
+        model = model_zoo.get(config_path, trained=True, device="cpu")
```

Comment:
I'm confused: does `device="cpu"` mean that it will always run on CPU? If so, then why are we skipping tests when cuda is not available? Even if cuda is available, we won't use it?

Reply:
Based on the Detectron2 documentation, if device is specified, it will overwrite the device specified in the documentation, so yes, I think it will load the model in cpu mode. I don't know if this was your doubt.

Reply:
I will try removing the CPU restriction and check whether the CI passes. If it does, I will remove the `device` argument